easybuild/easyblocks/p/pdt.py

```diff
@@ -14,37 +14,44 @@
 
 @author Bernd Mohr (Juelich Supercomputing Centre)
 @author Markus Geimer (Juelich Supercomputing Centre)
+@author Alexander Grund (TU Dresden)
 """
 
 import os
 
 from easybuild.easyblocks.generic.configuremake import ConfigureMake
 import easybuild.tools.toolchain as toolchain
 from easybuild.tools.build_log import EasyBuildError
-from easybuild.tools.run import run_cmd
 
 
-class EB_PDT(ConfigureMake):
-    """Support for building/installing PDT."""
+def find_arch_dir(install_dir):
+    """Find architecture directory inside the install tree
 
-    def __init__(self, *args, **kwargs):
-        """Initialisation of custom class variables for PDT."""
-        super(EB_PDT, self).__init__(*args, **kwargs)
+    For simplicity any top-level folder containing a bin and lib directory is collected
+    Raises an error if multiple or none matching folders are found
+    Returns the architecture specific folder
+    """
+    arch_dirs = []
+    for entry in os.listdir(install_dir):
+        full_path = os.path.join(install_dir, entry)
+        # Only check directories which are not symlinks
+        # E.g. on x86_64 there are symlinks craycnl, mic_linux, ... to x86_64
+        if not os.path.isdir(full_path) or os.path.islink(full_path):
+            continue
+        bin_dir = os.path.join(full_path, 'bin')
+        lib_dir = os.path.join(full_path, 'lib')
+        if all(os.path.isdir(dir) for dir in (bin_dir, lib_dir)):
+            arch_dirs.append(full_path)
 
-        out, _ = run_cmd("uname -m", simple=False)
-        self.machine = out.strip()
-        self.log.info("Using '%s' as machine label", self.machine)
+    if not arch_dirs:
+        raise EasyBuildError('Architecture specific directory not found in %s' % install_dir)
```

Comment:
```python
raise EasyBuildError("No architecture specific directories not found in %s", install_dir)
```

(mostly about the ` %` -> `,`; for strings containing spaces, we usually use double quotes)

Reply:
>  for strings containing spaces, we usually use double quotes

Hm? What is the reasoning?   
For reference: https://stackoverflow.com/a/144163/1930508 and I agree: Using conventions when to use which is confusing ;)

Reply:
It's more of a personal habit, we don't have strict code stye rules on this one.

The only way to avoid discussions like this is to adopt [Black](https://github.com/psf/black), but it's a significant effort and it formats some things in questionable/weird ways, which make me reluctant to switch to it.